### PR TITLE
limit number of suggestions in BubbleTextField suggestions dropdown, fix scrolling

### LINKED ISSUE #2949: Create overlay for buble text field suggestions
Right now the suggestions list from the bubble text field in mail editor changes the content size of the mail editor. This leads to not very nice movement of the content when typing in the mail address. We should make a real overlay for the suggestion list and also refactor the bubble text field input to the new component style.

### PR DESCRIPTION
**Pull request by vaf-hub**
fixes #2949

### CODE REVIEW

File: src/gui/base/BubbleTextField.js [hunk 1]
```diff
@@ -150,21 +158,33 @@ export class BubbleTextField<T> {
...
 				m(`.suggestions.abs.z4.full-width.elevated-bg.scroll.text-ellipsis${this.suggestions.length ? ".dropdown-shadow" : ""}`, {
-					oncreate: vnode => this._domSuggestions = vnode.dom,
+					oncreate: vnode => {
+						this._domSuggestions = vnode.dom
+						this._selectedSuggestionChangedListener = this.selectedSuggestion.map(
+							makeListSelectionChangedScrollHandler(this._domSuggestions, this.bubbleHandler.suggestionHeight, this._getSelectedSuggestionIndex.bind(this))
+						)
+					},
+					onbeforeremove: () => {
+						this._selectedSuggestionChangedListener.end()
```
Comment (johnbotris) [inline src/gui/base/BubbleTextField.js:168]: this should be `this._selectedSuggestionChangedListener.end(true)`

File: src/gui/base/BubbleTextField.js [hunk 2]
```diff
@@ -337,30 +360,30 @@ export class BubbleTextField<T> {
...
 		}
 		return false
 	}
 
 	handleDownArrow(): boolean {
-		if (this.selectedSuggestion != null) {
-			this.selectedSuggestion.selected = false
-			let next = (this.suggestions.indexOf(this.selectedSuggestion) + 1)
+		if (this.selectedSuggestion() != null) {
+			neverNull(this.selectedSuggestion()).selected = false
```
Comment (johnbotris) [inline src/gui/base/BubbleTextField.js:377]: I would personally rather put `this.selectedSuggestion()` in a variable first over using `neverNull`, here as well as further down

File: src/sharing/view/GroupSharingDialog.js [hunk 3]
```diff
@@ -182,7 +182,9 @@ function showAddParticipantDialog(model: GroupSharingModel, texts: GroupSharingT
 			})
 			return buttonAttrs
 		}
-	}, locator.contactModel))
+		// we hardcode the maximum number of BTF suggestions so the suggestions dropdown does not overflow the dialog
+	}, locator.contactModel, 3)
```
Comment (johnbotris) [inline src/sharing/view/GroupSharingDialog.js:186]: should use a constant here, avoid confusion later down the road (even with the comment)

**Review by johnbotris — CHANGES_REQUESTED** [review]

File: src/gui/base/BubbleTextField.js (hunk 1 above)
Comment (vaf-hub) [reply src/gui/base/BubbleTextField.js:168]: Sure. I will change it.

File: src/sharing/view/GroupSharingDialog.js (hunk 3 above)
Comment (vaf-hub) [reply src/sharing/view/GroupSharingDialog.js]: OK, I amended and pushed a commit.

File: src/gui/base/BubbleTextField.js (hunk 2 above)
Comment (vaf-hub) [reply src/gui/base/BubbleTextField.js]: Yeah, it's probably nicer. Amended and pushed the change.

**Review by johnbotris — APPROVED** [review]

**Pull request merged**